airflow/gcp/operators/campaign_manager.py

```diff
@@ -0,0 +1,347 @@
+    :type gcp_conn_id: str
+    :param delegate_to: The account to impersonate, if any. For this to work, the service accountmaking the
+        request must have  domain-wide delegation enabled.
+    :type delegate_to: str
+    """
+
+    template_fields = (
+        "profile_id",
+        "report_id",
+        "file_id",
+        "bucket_name",
+        "report_name",
+        "chunk_size",
+        "api_version",
+        "gcp_conn_id",
+        "delegate_to",
+    )
+
+    @apply_defaults
+    def __init__(  # pylint:disable=too-many-arguments
+        self,
+        profile_id: str,
+        report_id: str,
+        file_id: str,
+        bucket_name: str,
+        report_name: str,
+        gzip: bool = False,
+        chunk_size: int = 5 * 1024 * 1024,
+        api_version: str = "v3.3",
+        gcp_conn_id: str = "google_cloud_default",
+        delegate_to: Optional[str] = None,
+        *args,
+        **kwargs
+    ):
+        super().__init__(*args, **kwargs)
+        self.profile_id = profile_id
+        self.report_id = report_id
+        self.file_id = file_id
+        self.api_version = api_version
+        self.chunk_size = chunk_size
+        self.gzip = gzip
+        self.bucket_name = self._set_bucket_name(bucket_name)
+        self.report_name = self._set_report_name(report_name)
+        self.gcp_conn_id = gcp_conn_id
+        self.delegate_to = delegate_to
+
+    @staticmethod
+    def _set_report_name(name: str) -> str:
+        return name if name.endswith(".csv") else name + ".csv"
+
+    @staticmethod
+    def _set_bucket_name(name: str) -> str:
+        bucket = name if not name.startswith("gs://") else name[5:]
+        return bucket.strip("/")
+
+    def execute(self, context: Dict):
+        hook = GoogleCampaignManagerHook(
+            gcp_conn_id=self.gcp_conn_id,
+            delegate_to=self.delegate_to,
+            api_version=self.api_version,
+        )
+        gcs_hook = GoogleCloudStorageHook(
+            google_cloud_storage_conn_id=self.gcp_conn_id, delegate_to=self.delegate_to
+        )
+        self.log.info("Starting downloading report %s", self.report_id)
+        request = hook.get_report_file(
+            profile_id=self.profile_id, report_id=self.report_id, file_id=self.file_id
+        )
+
+        # Download the report
+        with tempfile.NamedTemporaryFile() as temp_file:
+            downloader = http.MediaIoBaseDownload(
+                fd=temp_file, request=request, chunksize=self.chunk_size
+            )
+            download_finished = False
+            while not download_finished:
+                _, download_finished = downloader.next_chunk()
+
+            temp_file.flush()
+            # Upload the local file to bucket
+            gcs_hook.upload(
+                bucket_name=self.bucket_name,
+                object_name=self.report_name,
+                gzip=self.gzip,
+                filename=temp_file.name,
+                mime_type="text/csv",
```

Comment:
Can we perform a multipart upload in order to support larger file sizes?
```suggestion
                mime_type="text/csv",
                multipart=True,
```

Reply:
Upload method fo the `GoogleCloudStorageHook` does not have this option.

Reply:
Ah, looks like this parameter was deprecated in 1.10.4.  OK as is.